internal/ingest/linksystem.go

```diff
@@ -449,8 +457,12 @@ func (ing *Ingester) indexContentBlock(adCid cid.Cid, pubID peer.ID, nentries ip
 	// entries CID to the ad CID so that the ad can be loaded when that chunk
 	// is received.
 	//
-	// Do not return here if error; try to index content in this chunk.
+	// If we error in reading the next link, then that means this entry chunk is
+	// malformed, so it is okay to skip it.
 	hasNextLink, linkErr := ing.setNextCidToAd(nchunk, adCid)
+	if linkErr != nil {
+		return linkErr
```

Comment:
```suggestion
	hasNextLink, err := ing.setNextCidToAd(nchunk, adCid)
	if err != nil {
		return err
```
OK, then just use `err` instead of `linkErr` since it is not being handled specially.